docs/current/guides/snippets/build-test-publish-php/dagger.php

```diff
@@ -0,0 +1,277 @@
+    // initialize client with
+    // endpoint from environment
+    $sessionPort = getenv('DAGGER_SESSION_PORT') or throw new Exception("DAGGER_SESSION_PORT doesn't exist");
+    $sessionToken = getenv('DAGGER_SESSION_TOKEN') or throw new Exception("DAGGER_SESSION_TOKEN doesn't exist");
+    $this->client = new Client(
+      'http://127.0.0.1:' . $sessionPort . '/query',
+      ['Authorization' => 'Basic ' . base64_encode($sessionToken . ':')]
+    );
+  }
+
+  // build base image
+  public function buildBaseImage() {
+    // get host working directory
+    $sourceQuery = <<<QUERY
+    query {
+      host {
+        directory (path: ".", exclude: ["vendor", "ci"]) {
+          id
+        }
+      }
+    }
+    QUERY;
+    $sourceDir = $this->executeQuery($sourceQuery);
+
+    // build runtime image
+    // install tools and PHP extensions
+    // configure Apache webserver root and rewriting
+    $runtimeQuery = <<<QUERY
+    query {
+      container {
+        from(address: "php:8.2-apache-buster") {
+          withExec(args: ["apt-get", "update"]) {
+            withExec(args: ["apt-get", "install", "--yes", "git-core"]) {
+              withExec(args: ["apt-get", "install", "--yes", "zip"]) {
+                withExec(args: ["docker-php-ext-install", "pdo", "pdo_mysql", "mysqli"]) {
+                  withExec(args: ["sh", "-c", "sed -ri -e 's!/var/www/html!/var/www/public!g' /etc/apache2/sites-available/*.conf"]) {
+                    withExec(args: ["sh", "-c", "sed -ri -e 's!/var/www/!/var/www/public!g' /etc/apache2/apache2.conf /etc/apache2/conf-available/*.conf"]) {
+                      withExec(args: ["a2enmod", "rewrite"]) {
+                        id
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    QUERY;
+    $runtime = $this->executeQuery($runtimeQuery);
+
+    // add application source code
+    // set file permissions
+    // set environment variables
+    $appQuery = <<<QUERY
+    query {
+      container (id: "$runtime") {
+        withMountedDirectory(path: "/mnt", source: "$sourceDir") {
+          withWorkdir(path: "/mnt") {
+            withExec(args: ["cp", "-R", ".", "/var/www"]) {
+              withExec(args: ["chown", "-R", "www-data:www-data", "/var/www"]) {
+                withExec(args: ["chmod", "-R", "777", "/var/www/storage"]) {
+                  withExec(args: ["chmod", "+x", "/var/www/docker-entrypoint.sh"]) {
+                    id
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    QUERY;
+    $app = $this->executeQuery($appQuery);
+
+    // install Composer
+    // add application dependencies
+    $appWithDepsQuery = <<<QUERY
+    query {
+      container (id: "$app") {
+        withExec(args: ["sh", "-c", "curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer"]) {
```

Comment:
If we use `curl`, we should install the package explicitly & not assume that it exists.